### PR TITLE
PORI-231: Events base theme and header & footer theming.

### PR DESCRIPTION
**Pull request by tuukkaturu**
Events base theme and header & footer theming.

To test:
1. first you need to have working pori-events local setup.
2. drush cim -y && drush cr
3. make sure pori_events theme is enabled.
4. go to /admin/structure/block/list/pori_events there you should have 3 blocks in footer region
5. you can provide content for the blocks in custom blocks tab in the block menu
6. go back to front page and confirm that there is a logo in header and footer, footer blocks should show the text.
7. Slogan is provided from template file hardcoded, since the text is too long for the actual slogan field in site settings. Slogan is hidden in mobile for now since there is no menu implemented yet, also language-switcher does not exist yet.

### CODE REVIEW

File: drupal/web/themes/custom/pori_events/src/scss/base/_typography.scss [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    text-decoration: none;
+  }
+}
+
+h1 {
+  font-size: $font-size-heading-1;
+  line-height: 1.1666;
+  margin-bottom: 3.15rem;
+
+  @media screen and (max-width: 599px) {
```
Comment (teroelonen) [inline drupal/web/themes/custom/pori_events/src/scss/base/_typography.scss]: Media-queries would be good to set up in a separate file and with variables.

File: drupal/web/themes/custom/pori_events/src/scss/components/block/_branding.scss [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    padding: 1.5rem 0;
+  }
+}
+
+.site-slogan {
+  display: none;
+  font-family: $heading-font-family;
+  font-size: .85rem;
+  letter-spacing: -0.5px;
+  padding: .5rem 0;
```
Comment (teroelonen) [inline drupal/web/themes/custom/pori_events/src/scss/components/block/_branding.scss]: Alphabetic order.

File: drupal/web/themes/custom/pori_events/src/scss/lib/mixins/_fa.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/// Include a Font Awesome Icon
+/// @param {string} $content [none] - content icon ("\000")
+/// @param {number} $font-size [16] - font size to scale the icon, rem unit
+@mixin fa($content, $font-size: 1rem, $line-height: 1, $responsive: $default-responsive-font-size) {
+  @include fontsmoothing();
+  color: inherit;
+  content: $content;
+  font-family: FontAwesome;
+  @if $responsive {
```
Comment (teroelonen) [inline drupal/web/themes/custom/pori_events/src/scss/lib/mixins/_fa.scss]: Empty line before this.

File: drupal/web/themes/custom/pori_events/src/scss/lib/mixins/_typography.scss [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  @if $uses-text-shadow {
+    text-shadow: 1px 1px 1px rgba(0, 0, 0, .004);
+  }
+}
+
+@mixin font-size-responsive($size) {
+  font-size: $size;
+
+  @media screen and (max-width: 599px) {
```
Comment (teroelonen) [inline drupal/web/themes/custom/pori_events/src/scss/lib/mixins/_typography.scss]: Here as well, maybe use a variable for the breakpoint?

**Comment by teroelonen** [comment]
screen shot 2018-01-19 at 12 53 33
screen shot 2018-01-19 at 12 49 31

File: drupal/web/themes/custom/pori_events/src/scss/base/_typography.scss (hunk 1 above)
Comment (tuukkaturu) [reply drupal/web/themes/custom/pori_events/src/scss/base/_typography.scss]: True, i took this straight from kada theme, seems to wrong in there. I'll change this.

File: drupal/web/themes/custom/pori_events/src/scss/lib/mixins/_typography.scss (hunk 4 above)
Comment (tuukkaturu) [reply drupal/web/themes/custom/pori_events/src/scss/lib/mixins/_typography.scss]: True, i took this straight from kada theme, seems to wrong in there. I'll change this.

**Comment by tuukkaturu** [comment]
Adjusted the logo width a bit so header logo is not so big on wider screen and footer logo should fit in smallest screen.

Apparently custom blocks dont work how i remembered. Will need to think another way to do the footer content.

**Comment by tuukkaturu** [comment]
Instead of custom blocks, created footer-information content-type and view to display the content.
New master branch merged so composer update --with-dependencies might be needed to test again.
To test:
1. composer update --with-dependencies
2. drush cim -y && drush cr (you might need to run this few times, sometimes there is issues importing configs)
3. go to add content and add footer information, add something to the first 3 sections, fourth is there just in case there is a need
4. go to front page and confirm that content is displayed.

**Pull request merged**